### ISSUE TITLE
Change the implementation of Core/Engine/Rss away from spoon library

### ISSUE THREAD

**Issue by carakas** [forkcms/forkcms#2098]
## Type

- Enhancement

## Problem description

We are using Spoon rss, since spoon is deprecated we should swap it out

## Expected behavior / Proposed solutions
Keep the same external api but let it work on a different library or implementation

**Comment by carakas** [comment]
Suggestions welcome on what we should use instead

**Comment by jonasdekeukelaere** [comment]
I don't know any bundles or symfony stuff for that. Maybe an XML writer?

**Comment by tijsverkoyen** [comment]
https://github.com/alexdebril/rss-atom-bundle? 
Or just some twig-template?

**Comment by carakas** [comment]
if this can be implemented in a BC way it can be done in 5.1 else it wil be 6

**Comment by jeroendesloovere** [comment]
https://github.com/alexdebril/rss-atom-bundle seems indeed like a nice one!
It uses `providers`.
So we can for example create a `.../Blog/Provider/BlogRssProvider.php`.

**Comment by tijsverkoyen** [comment]
I don't agree at all. A module should be responsible for its own feeds, not a global RSS-feed.

**Comment by jeroendesloovere** [comment]
@tijsverkoyen nobody needs a global RSS-feed indeed... Don't know why you say this...

Some more information about RSS Atom bundle:
* There is a controller that can generate a RSS feed.
* You just have to throw a `provider` at it, that will automatically get your *items* to generate *rss feed*.
```
    <route id="your_route_name" pattern="/your/route/{contentId}">
        <default key="_controller">DebrilRssAtomBundle:Stream:index</default>
        <default key="format">rss</default>
        <default key="source">your.provider.service</default>
    </route>
```
> `your.provider.service` can be for example `BlogArticleFeedProvider` or `FaqQuestionFeedProvider`, ...

** Provider **
In the `getFeedContent()` method you must return a `FeedInterface`.

Documentation:
* [Providing feeds](https://github.com/alexdebril/rss-atom-bundle#providing-feeds)
* [Choosing your own provider](https://github.com/alexdebril/rss-atom-bundle#choosing-your-own-provider)
